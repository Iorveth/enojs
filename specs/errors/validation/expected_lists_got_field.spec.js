const eno = require('../../../eno.js');

const input = `
languages:
| eno (eno notation)
| yaml (yaml ain't markup language)

languages:
| json (javascript object notation)
| cson (coffeescript object notation)
`.trim();

describe('validation.expectedListsGotField', () => {
  const document = eno.parse(input);

  let error;
  try {
    document.lists('languages');
  } catch(err) {
    error = err;
  }

  it(`provides a correct message`, () => {
    expect(error.message).toMatchSnapshot();
  });

  it(`provides correct selection metadata`, () => {
    expect(error.selection).toMatchSnapshot();
  });
});
