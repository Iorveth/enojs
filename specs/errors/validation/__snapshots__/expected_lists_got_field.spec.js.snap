// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation.expectedListsGotField provides a correct message 1`] = `
"Only lists with the name 'languages' were expected, but a field with this name was found.

   Line | Content
 >    1 | languages:
 >    2 | | eno (eno notation)
 >    3 | | yaml (yaml ain't markup language)
      4 | 
      5 | languages:
   ...
"
`;

exports[`validation.expectedListsGotField provides correct selection metadata 1`] = `
Array [
  Array [
    0,
    0,
  ],
  Array [
    2,
    35,
  ],
]
`;
